packages/eslint-plugin/src/rules/shorthand-property-sorting/index.ts

```diff
@@ -0,0 +1,128 @@
+import {
+  shorthandBuckets,
+  shorthandFor,
+  kebabCase,
+  type ShorthandProperties,
+} from '@compiled/utils';
+import type { Rule, Scope } from 'eslint';
+import type { ObjectExpression, Property } from 'estree';
+
+import { isCss, isCssMap, isStyled } from '../../utils';
+
+const arePropertiesInTheRightOrder = (
+  node: ObjectExpression,
+  propertyA: Property,
+  i: number
+): boolean => {
+  if (node.properties.length > 0 && propertyA.key.type === 'Identifier') {
+    const propA = kebabCase(propertyA.key.name) as ShorthandProperties;
+    const depthA = shorthandBuckets[propA];
+
+    for (let j = i + 1; j < node.properties.length; j++) {
+      const propertyB = node.properties[j];
+
+      if (propertyB.type === 'Property') {
+        if (propertyB.key.type === 'Identifier') {
+          const propB = kebabCase(propertyB.key.name) as ShorthandProperties;
+          const depthB = shorthandBuckets[propB];
+
+          const shorthandForResA = shorthandFor[propA];
+          const shorthandForResB = shorthandFor[propB];
+
+          if (Array.isArray(shorthandForResA) && Array.isArray(shorthandForResB)) {
+            // find intersection between objects
+            const intersectionAB = shorthandForResA.filter((x) => shorthandForResB.includes(x));
+
+            if (intersectionAB.length > 0 && depthB < depthA) {
+              return true;
+            }
+          }
+        }
+      }
+    }
+  }
+  return false;
+};
+
+const callExpressionCorrectImport = (node: Rule.Node, references: Scope.Reference[]): boolean => {
+  return (
+    node.type === 'CallExpression' &&
+    (isCss(node.callee as Rule.Node, references) ||
+      isCssMap(node.callee as Rule.Node, references) ||
+      isStyled(node.callee as Rule.Node, references))
+  );
+};
+
+export const shorthandFirst: Rule.RuleModule = {
+  meta: {
+    docs: {
+      recommended: true,
+      description:
```

Comment:
Can we change the description to reflect what is in the [README](https://github.com/atlassian-labs/compiled/pull/1714/files#diff-9c805949ce856aceb094bd27cd205e3341156dee6f13f17566c9035cfa494c58R5)

Reply:
Is the description necessary? I have the url which links directly to the README. I don't want to keep squeezing things into one string it looks messy

Reply:
If that's the case, I would shorten the description to something more descriptive and link the README, for example

Prevent unwanted side-effect by ensuring shorthand properties are always defined before their related longhands. See more in the README.

So you have only 1 thing to maintain